### PR TITLE
Try new icons for reusable blocks and template parts.

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

Starts work on #33897 by adding new generic icons for template parts and reusable blocks.

For now, the PR adds the two new icons, and applies them to template parts and partially, reusable blocks:

Screenshot 2021-08-11 at 11 01 42

As shown above, there's still some logic in the Reusable Block system that falls back to the `defaultBlock` icon, as shown above:

- In the List View
- In the Inspector description

Additionally, the `defaultBlock` is used for select mode:

Screenshot 2021-08-11 at 11 01 54

If you're familiar with the reusable blocks codebase, I'd appreciate your help in applying the new icon in those remaining place. 

## How has this been tested?

Template Parts and Reusable blocks, when used in the canvas, list view, and inspector, should leverage the new icons.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]

File: packages/block-editor/src/store/selectors.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ import {
 	getPossibleBlockTransformations,
 	parse,
 } from '@wordpress/blocks';
-import { SVG, Rect, G, Path } from '@wordpress/components';
 import { Platform } from '@wordpress/element';
+import { reusableBlockType } from '@wordpress/icons';
```
Comment (mtias) [inline packages/block-editor/src/store/selectors.js]: I think maybe we should call this one "symbol" and "symbol-filled" at the icon level

**Comment by jasmussen** [comment]
Thanks for the fix Nik, thanks for the review! I'll address Matías' comment and land this!

**Comment by jasmussen** [comment]
Fixes #30338, fixes #33897.

**Comment by jasmussen** [comment]
Renamed the icons, and things are looking good. Feel free to press the green button when the checks pass, I have to leave in a little bit.

Screenshot 2021-08-11 at 12 24 04

**Comment by jasmussen** [comment]
Thank you all for the help landing this! 👌

**Comment by Addison-Stavlo** [comment]
👋   Im a bit confused on the Icon choice for "Template Part" here.  Previously, the icons for template parts and their variations were very coherent and cohesive as a group.  Header, Footer, and Sidebar icons are all of the same form with a different area filled in to distinguish them, while the generic "Template Part" icon was the same with no area filled in to distinguish the genericness of the non-variation version.

Now, we have one icon for the generic version that has no relation whatsoever to the icons for its variations:

Screen Shot 2021-08-23 at 8 16 38 AM

**Comment by jasmussen** [comment]
> Previously, the icons for template parts and their variations were very coherent and cohesive as a group. Header, Footer, and Sidebar icons are all of the same form with a different area filled in to distinguish them, while the generic "Template Part" icon was the same with no area filled in to distinguish the genericness of the non-variation version.

In nesting contexts the previous icon was reported as [confusing](https://github.com/WordPress/gutenberg/issues/30338) as its design suggested a particular layout that might not be leveraged by that particular generic template part at all.

The new icon is intentionally generic as discussed in https://github.com/WordPress/gutenberg/issues/33897 so as to clearly denote its special linked nature when shown in the list view. 

Your comment made me realize a bug, though — the generic icon is meant to _also_ be used for header, footer and sidebar template parts when shown in the list view:



But that doesn't appear to be the case currently:

Screenshot 2021-08-24 at 09 50 10

@ntsekouras can you help me make that happen?

**Comment by ntsekouras** [comment]
>Your comment made me realize a bug, though — the generic icon is meant to also be used for header, footer and sidebar template parts when shown in the list view:

@jasmussen do you mean to remove the `header, footer` icons completely, or just to show the generic icon in `list view` and in other places like the `toolbar` and the `block card` to show the existing icons?

**Comment by jasmussen** [comment]
The latter, show the generic icon in toolbar, block card and list views. The icons still feel useful in cases like the area selector dropdown that's being proposed in #29295:

128333827-7642d66a-e0ec-4268-8f9b-627519830d31

**Comment by jasmussen** [comment]
@ntsekouras just to make sure https://github.com/WordPress/gutenberg/pull/34002#issuecomment-904409098 doesn't fall off the radar, should I open a separate ticket for it? I'd be happy to pair on it.

**Comment by jasmussen** [comment]
I went ahead and ticketed it in #35453.

**Pull request merged**